lisa/analysis/pixel6.py

```diff
@@ -0,0 +1,100 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import pandas as pd
+import holoviews as hv
+from holoviews import opts
+
+from lisa.datautils import df_refit_index
+from lisa.analysis.base import TraceAnalysisBase
+from lisa.trace import requires_events
+from lisa.notebook import plot_signal
+
+P6_EMETER_CHAN_NAMES = {
+    'S4M_VDD_CPUCL0': 'little',
+    'S3M_VDD_CPUCL1': 'mid',
+    'S2M_VDD_CPUCL2': 'big',
+}
+
+class Pixel6Analysis(TraceAnalysisBase):
+    """
+    Support for Pixel 6-specific analysis data
+
+    :param trace: input Trace object
+    :type trace: :class:`trace.Trace`
+    """
+
+    name = 'pixel6'
+
+###############################################################################
+# DataFrame Getter Methods
+###############################################################################
+    @TraceAnalysisBase.cache
+    @requires_events('pixel6_emeter')
+    def df_power_meter(self):
+        """
+        Get the power meter readings across the trace.
+
+        :retuns: A :class:`pandas.DataFrame` with:
+            * A ``little`` column (readings from the little cluster)
+            * A ``mid`` column (readings from the mid cluster)
+            * A ``big`` column (readings from the big cluster)
+        """
+        df = self.trace.df_event('pixel6_emeter')
+        df = df.pivot_table(values='value', index='ts', columns='chan_name').reset_index().rename(columns=P6_EMETER_CHAN_NAMES)
+        channels = list(P6_EMETER_CHAN_NAMES.values())
+        df['ts_diff'] = df['ts'].diff()
+        # convert channels to power usage delta in mW
+        for channel in channels:
+            df[channel] = df[channel].diff() / df["ts_diff"]
+        # convert ts to seconds
+        df['Time'] = df['ts'] / 1000
+        df = df[['Time'] + channels].set_index('Time').iloc[1:]
+        return df
+
+###############################################################################
+# Plotting methods
+###############################################################################
+
+    @TraceAnalysisBase.plot_method
+    @df_power_meter.used_events
+    def plot_power_meter(self, channels=None):
+        """
+        Plot the power meter readings from various channels.
+
+        :param channels: List of channels to plot
+        :type channels: list(str)
+
+        The channels needs to correspond to columns in df_power_meter().
+        """
+        df = self.df_power_meter()
+
+        channels = channels if channels is not None else list(df.columns)
+        if any([channel not in df.columns for channel in channels]):
+            raise ValueError('Specified channel not found')
+        idx = df.index.name
+
+
+        ds = hv.Dataset(df, idx, channels)
```

Comment:
you should be able to create the Curve directly without making an explicit Dataset. You might want to try the `plot_signal` function for something like that as it adds some stuff and makes it easier to maintain consistent plots in all analysis